pkg/operator/controllers/checkers/ingresscertificatechecker/checker_test.go

```diff
@@ -31,37 +34,101 @@ func TestCheck(t *testing.T) {
 		name              string
 		ingressController *operatorv1.IngressController
 		clusterVersion    *configv1.ClusterVersion
+		cluster           *arov1alpha1.Cluster
 		wantErr           string
 	}{
 		{
-			name:           "default certificate is set",
+			name:    "shoud return clusterVersion not found error when ClusterVersion is nil",
+			wantErr: `clusterversions.config.openshift.io "version" not found`,
+		},
+		{
+			name:              "shoud return IngressController not found error when IngressController is nil",
+			ingressController: nil,
+			clusterVersion:    clusterVersion,
+			wantErr:           `ingresscontrollers.operator.openshift.io "default" not found`,
+		},
+		{
+			name:              "should return Cluster not found error when Cluster is nil",
+			clusterVersion:    clusterVersion,
+			ingressController: fakeIngressController(nil),
+			cluster:           nil,
+			wantErr:           "clusters.aro.openshift.io \"cluster\" not found",
+		},
+		{
+			name:              "should return certificate error (ARO's fault) when no certificate is set and we have a managed domain",
+			clusterVersion:    clusterVersion,
+			ingressController: fakeIngressController(nil),
+			cluster:           fakeCluster("whatever.aroapp.io"),
+			wantErr:           "missing ingress certificate for cluster with managed domain",
+		},
+		{
+			name:              "should return error (customer's fault) when no certificate is set and we have a custom domain",
+			clusterVersion:    clusterVersion,
+			ingressController: fakeIngressController(nil),
+			cluster:           fakeCluster("whatever.domain.io"),
+			wantErr:           "missing ingress certificate for cluster with custom domain",
+		},
+		{
+			name:           "should return error when we have a managed domain and we have an invalid certificate name",
+			clusterVersion: clusterVersion,
+			ingressController: fakeIngressController(&corev1.LocalObjectReference{
+				Name: "fake-custom-name-ingress",
+			}),
+			cluster: fakeCluster("whatever.aroapp.io"),
+			wantErr: `invalid ingress certificate name for cluster with managed domain`,
```

Comment:
why are your string not always with `"` ?

Reply:
` was already there, I just changed the content of the string, so I forgot to change it to ", doing it now.
Tks!